integration/virtual_kubelet.go

```diff
@@ -0,0 +1,94 @@
+package integration
+
+import (
+	"io"
+	"net/url"
+	"os"
+	"path"
+	"time"
+
+	"github.com/kubernetes-sig-testing/frameworks/integration/internal"
+)
+
+type VirtualKubelet struct {
+	URL          *url.URL
+	Path         string
+	Args         []string
+	ConfDir      string
+	StartTimeout time.Duration
+	StopTimeout  time.Duration
+	Out          io.Writer
+	Err          io.Writer
+
+	Conf string
+
+	processState *internal.ProcessState
+}
+
+func (vk *VirtualKubelet) Start() error {
+	var err error
+
+	vk.processState = &internal.ProcessState{}
+
+	vk.processState.DefaultedProcessInput, err = internal.DoDefaulting(
+		"virtual-kubelet",
+		vk.URL,
+		vk.ConfDir,
+		true,
+		vk.Path,
+		vk.StartTimeout,
+		vk.StopTimeout,
+	)
+	if err != nil {
+		return err
+	}
+
+	vk.processState.StartMessage = "Node 'virtual-kubelet' with OS type 'Linux' registered"
```

Comment:
is this going to work on non-linux? Does it matter?

Reply:
The `virtual-kubelet` with the mock provider [identifies itself as a linux node](https://github.com/virtual-kubelet/virtual-kubelet/blob/a99f2d08c1f57e39b8d5e65ebf70b82ee000a9ce/providers/mock/mock.go#L245-L247). Also we develop on Darwin mostly and can confirm that this *works on my machine* ;)